src/sys/windows.rs

```diff
@@ -735,6 +736,52 @@ pub(crate) fn from_in6_addr(addr: in6_addr) -> Ipv6Addr {
     Ipv6Addr::from(*unsafe { addr.u.Byte() })
 }
 
+/// This type is not yet in winapi.
+#[repr(C)]
+#[allow(non_camel_case_types)]
+struct sockaddr_un {
+    pub sun_family: sa_family_t,
+    pub sun_path: [u8; 108],
+}
+
+#[allow(unused_unsafe)] // TODO: replace with `unsafe_op_in_unsafe_fn` once stable.
+pub(crate) fn unix_sockaddr(path: &Path) -> io::Result<SockAddr> {
+    unsafe {
+        SockAddr::init(|storage, len| {
+            // Safety: `SockAddr::init` zeros the address, which is a valid
+            // representation.
+            let storage: &mut crate::sys::sockaddr_un = unsafe { &mut *storage.cast() };
+            let len: &mut socklen_t = unsafe { &mut *len };
+
+            let bytes = path
+                .to_str()
+                .ok_or_else(|| {
+                    io::Error::new(io::ErrorKind::InvalidInput, "path must be valid UTF-8")
+                })?
+                .as_bytes();
+
+            if bytes.len() > storage.sun_path.len() {
+                return Err(io::Error::new(
+                    io::ErrorKind::InvalidInput,
+                    "path must be shorter than SUN_LEN",
+                ));
+            }
+
+            storage.sun_family = crate::sys::AF_UNIX as sa_family_t;
+            storage.sun_path[..bytes.len()].copy_from_slice(bytes);
+
+            let base = storage as *const _ as usize;
+            let path = &storage.sun_path as *const _ as usize;
+            let sun_path_offset = path - base;
+            let length = sun_path_offset + bytes.len();
```

Comment:
I think this needs +1 for the null terminator, see https://github.com/rust-lang/socket2/blob/1c67209c2eda5708b3093900a59befbc4a981b13/src/sys/unix.rs#L490-L495

Reply:
The Windows implementation doesn't require null termination as long as the length is set appropriately. I can include this for consistency if you'd like, but I don't think it's strictly necessary.

Reply:
See comment above, not necessary.